packages/ckeditor5-engine/src/view/observer/inputobserver.js

```diff
@@ -24,21 +26,98 @@ export default class InputObserver extends DomEventObserver {
 	}
 
 	onDomEvent( domEvent ) {
-		this.fire( domEvent.type, domEvent );
+		const domTargetRanges = Array.from( domEvent.getTargetRanges() );
```

Comment:
Is an iterable always returned from `getTargetRanges`? 

BTW Spec says it returns array so maybe we don't need `Array.from()`?

Reply:
[You're right](https://www.w3.org/TR/input-events-2/#dom-inputevent-gettargetranges) :+1: [](https://www.w3.org/TR/input-events-2/#dom-inputevent-gettargetranges)